src/Facets/RelayDepositoryFacet.sol

```diff
@@ -0,0 +1,148 @@
+        RelayDepositoryData calldata _relayDepositoryData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _relayDepositoryData);
+    }
+
+    /// @notice Performs a swap before depositing into Relay Depository
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _relayDepositoryData Data specific to Relay Depository
+    function swapAndStartBridgeTokensViaRelayDepository(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        RelayDepositoryData calldata _relayDepositoryData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _relayDepositoryData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for depositing into Relay Depository
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _relayDepositoryData Data specific to Relay Depository
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        RelayDepositoryData calldata _relayDepositoryData
+    ) internal {
+        // Validate that the provided depository matches our immutable address
+        if (_relayDepositoryData.depository != RELAY_DEPOSITORY) {
+            revert InvalidCallData();
+        }
+
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            // Native token deposit
+            IRelayDepository(_relayDepositoryData.depository).depositNative{
+                value: _bridgeData.minAmount
+            }(msg.sender, _relayDepositoryData.orderId);
+        } else {
+            // ERC20 token deposit
+            LibAsset.maxApproveERC20(
+                IERC20(_bridgeData.sendingAssetId),
+                _relayDepositoryData.depository,
+                _bridgeData.minAmount
+            );
+
+            IRelayDepository(_relayDepositoryData.depository).depositErc20(
+                msg.sender,
+                _bridgeData.sendingAssetId,
+                _bridgeData.minAmount,
+                _relayDepositoryData.orderId
+            );
+        }
+
+        emit RelayDepositoryDeposit(
```

Comment:
what's the purpose of RelayDepositoryDeposit event?
depository contract already emits RelayErc20Deposit or RelayNativeDeposit

Reply:
This can be removed I think

Reply:
correct, I removed it